test/plugin/test_out_google_cloud.rb

```diff
@@ -240,10 +244,16 @@ def http_request_message
     HTTP_REQUEST_MESSAGE
   end
 
-  # A wrapper around the constant HTTP_REQUEST_MESSAGE_WITHOUT_REFERER, so the
-  # definition can be skipped in the shared module and defined here.
-  def http_request_message_without_referer
-    HTTP_REQUEST_MESSAGE_WITHOUT_REFERER
+  # A wrapper around the constant HTTP_REQUEST_MESSAGE_EXPECTED_WITHOUT_REFERER,
+  # so the definition can be skipped in the shared module and defined here.
+  def http_request_message_expected_without_referer
```

Comment:
Refactoring has been done to implement the default `http_request_message` method in `base_test`, and let the grpc test override them. And a non-virtual `http_request_message_with_nil_referer` has been added.